### PR TITLE
[caprisun] Improvements to software-defined assets support in Dagit

### PR DESCRIPTION
**Pull request by bengotow**
## Summary
This PR includes a variety of improvements designed to blend software defined assets into the jobs UI and the asset catalog.

The big wins here are:
- The left nav is still jobs, but if you click an asset job you get the asset graph INSTEAD of the op graph in the overview tab. Clicking assets shows you a blended static + materializations sidebar.
- The asset graph styles foreign / external asset links more cleanly and supports tags, no longer uses color dots to indicate data freshness, and renders descriptions in plaintext instead of un-rendered markdown. Empty / unexecuted nodes also look less like the foreign links and have a blank space where the materialization info belongs.
- The asset catalog now mixes materializations and SDAs together and provides descriptions and more context. 

image
image

98480bc4 Make dbt tag a better color
72d4d416 New graphql resolvers for retrieving single asset nodes, super basic description on detail page
bc3bd3a9 Don't allow clicks to show a modal with metadata if the only entry is `kind`
5c388003 add compute kind to asset defs (#4944)
8d0deb03 Tags in the asset graph, static definitions in the asset catalog list page
91624215 Make job overview load workspace assets to determine if pipeline is an asset job
21162340 Try out a very small "upstream changed" label
fded2625 Add link to asset catalog to the materializations section, dividing line
f24be910 Show a "stale" tag instead of the status dot for now
2bfde2d8 Clean up asset graph, use in place of op graph for "dbt" job (hardcoded right now)

## Test Plan
Changes manually tested

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by sryza** [comment]
We've already chatted about these changes at length, and I'm quite excited about them.  A few small things that I noticed:
* Instead of "Static Definition", we can just say "Definition"
* The definition information appears to be repeated underneath the "Materialization in Last Run" section
* Want to make sure that the new Asset Catalog view only shows up if the flag is turned on? Otherwise, existing users will get columns that are always empty.

**Review by sryza — APPROVED** [review]
Can't comment on the implementation, but the outcome LGTM!

File: js_modules/dagit/packages/core/src/workspace/asset-graph/AssetGraphExplorer.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  onChangeExplorerPath: (path: PipelineExplorerPath, mode: 'replace' | 'push') => void;
+}
+
+export const AssetGraphExplorer: React.FC<Props> = (props) => {
+  const {repoAddress, handles, selectedHandle, explorerPath, onChangeExplorerPath} = props;
+  const repositorySelector = repoAddressToSelector(repoAddress);
+  const queryResult = useQuery<AssetGraphQuery, AssetGraphQueryVariables>(ASSETS_GRAPH_QUERY, {
+    variables: {repositorySelector},
+    notifyOnNetworkStatusChange: true,
+    pollInterval: POLL_INTERVAL,
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/workspace/asset-graph/AssetGraphExplorer.tsx:54]: Is it important to poll this?

File: js_modules/dagit/packages/core/src/workspace/asset-graph/AssetNode.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      handleLaunchResult(basePath, definition.jobName, result, true);
+    } catch (error) {
+      showLaunchError(error as Error);
+    }
+  };
+
+  return (
+    <ContextMenu
+      content={
+        <Menu>
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/workspace/asset-graph/AssetNode.tsx:73]: Can it use `MenuWIP` and `MenuItemWIP`?

File: js_modules/dagit/packages/core/src/workspace/asset-graph/ForeignNode.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import {ColorsWIP} from '../../ui/Colors';
+import {IconWIP} from '../../ui/Icon';
+import {FontFamily} from '../../ui/styles';
+
+import {assetKeyToString} from './Utils';
+
+export const ForeignNode: React.FC<{assetKey: {path: string[]}}> = ({assetKey}) => (
+  <ForeignNodeLink>
+    <span className="label">{assetKeyToString(assetKey)}</span>
+    <IconWIP name="open_in_new" color={ColorsWIP.Gray500} size={16} />
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/workspace/asset-graph/ForeignNode.tsx:13]: 16 is the default size

**Review by hellendag — APPROVED** [review]

File: js_modules/dagit/packages/core/src/workspace/asset-graph/ForeignNode.tsx (hunk 3 above)
Comment (bengotow) [reply js_modules/dagit/packages/core/src/workspace/asset-graph/ForeignNode.tsx:13]: Oh nice will fix this!

File: js_modules/dagit/packages/core/src/workspace/asset-graph/AssetGraphExplorer.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagit/packages/core/src/workspace/asset-graph/AssetGraphExplorer.tsx:54]: Hmm good catch, I think we can rm this - it looks like the non-asset version of this explorer doesn't poll so this doesn't need to either.

**Pull request merged**